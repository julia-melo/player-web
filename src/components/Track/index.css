.track {
  display: flex;
  align-items: center;

  width: 100%;
  height: 100%;
}

.track .track-item {
  padding-right: 12px;
  padding-left: 12px;
}

.track .arrow-right-icon {
  margin-left: auto;
}