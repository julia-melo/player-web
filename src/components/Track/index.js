import React from 'react'
import PropTypes from 'prop-types'

import { ArrowRightIcon, ArrowLeftIcon } from '../Icons'

import './index.css'

const Track = ({ itensList, children }) => {
  return (
    <div className='track'>
      <ArrowLeftIcon width={32} height={32} disabled />
      {itensList.map(item => (
        <div
          key={`track-${item.id}`}
          className='track-item'
        >
          {children(item)}
        </div>
      ))}
      <ArrowRightIcon width={32} height={32} />
    </div>
  )
}

Track.propTypes = {
  itensList: PropTypes.array.isRequired,
  children: PropTypes.func.isRequired,
}

export default Track