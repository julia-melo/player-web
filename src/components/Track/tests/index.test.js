import React from 'react'
import renderer from 'react-test-renderer'

import Track from '../index'

describe('Track component', () => {
  it('renders correctly', () => {
    const itensListMock = ['item example']
    const tree = renderer
      .create(<Track itensList={itensListMock}>{item => item}</Track>)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})