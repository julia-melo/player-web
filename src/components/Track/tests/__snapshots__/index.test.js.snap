// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Track component renders correctly 1`] = `
<div
  className="track"
>
  <div
    className="arrow-left-icon disabled "
  />
  <div
    className="track-item"
  >
    item example
  </div>
  <div
    className="arrow-right-icon  "
  />
</div>
`;
